lib/publiccloud/provider.pm

```diff
@@ -343,6 +343,14 @@ C<instance_type> defines the flavor of the instance. If not specified, it will l
 =cut
 
 sub create_instance {
+    my $max = get_var('PUBLIC_CLOUD_CREATE_INSTANCE_MAX_CALLS', 1);
+    if ($max > 0) {
+        my $count = get_var('_PUBLIC_CLOUD_CREATE_INSTANCE_CALL_COUNT', 0);
```

Comment:
Why is this handled as a openQA variable and not simply as a Perl variable? :thinking:

Reply:
Would a global persist state across different test modules? I was not sure about that.

Reply:
+1 on the goal being to limit this across all test modules. I don't think there's an easy way to do that via only perl.

Reply:
> Why is this handled as a openQA variable and not simply as a Perl variable?

because perl variable will not survive call from different module

Reply:
By using the `our` keyword you can create a package-level variable that can be used within a single test run as a single counter. We are doing something similar already in https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/instances.pm#L15

Reply:
`our $instance_counter` should do the trick IMHO.

Reply:
Indeed, we have the same intra-modules passing issue with `args`.
Could this one be handled similarly, by creating upper global perl variable to pass, instead of a new `yaop` (yet-another-opeqa-parameter) :slightly_smiling_face: ?

Reply:
TIL "our". Very cool, thanks! Fixed in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/20721/commits/342026f3c9f2fd4ce1efbf3339f66701d3248659

Reply:
> our $instance_counter should do the trick IMHO. 

AFAIK it does not . each module is loaded as separate perl process in backend . they technically unreachable except openQA variable , no matter which cool perl trick you will use .

Reply:
I confirmed it here http://rmarliere-openqa.qe.prg2.suse.org/tests/197#step/prepare_instance/196

Reply:
here we go https://github.com/os-autoinst/os-autoinst/blob/master/autotest.pm#L150 `os-autoinst` is doing `eval` of each `.pm` file independently . so all `ours` of one pm will end up with it

Reply:
Ah yes my test is wrong because I added the extra call within the same module...

Reply:
This time I added an extra call to registration.pm: http://rmarliere-openqa.qe.prg2.suse.org/tests/198#step/registration/2

The counter value was saved across modules. Using "our" is viable.